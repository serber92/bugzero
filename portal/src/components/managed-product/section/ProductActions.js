import React from 'react';
import { useHistory } from 'react-router-dom';

import PropTypes from 'prop-types';
import LoaderButton from '../../loader-button/LoaderButton';
import { Col, Button, Row, FormGroup } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTrash } from '@fortawesome/free-solid-svg-icons';

export default function ProductActions({ isLoading, handleDelete, handleSave, handleSaveAndAdd, vendorId, productId }) {
  const history = useHistory();

  return (
    <>
      <Row className="flex-between-center" style={{ marginTop: '40px' }}>
        <Col xs="4">
          <FormGroup>
            <Button block color="falcon-danger" onClick={() => history.push(`/vendors/${vendorId}/products`)}>
              Cancel
            </Button>
          </FormGroup>
        </Col>
        <Col xs="4">
          <FormGroup>
            {vendorId === 'hpe' ? (
              <LoaderButton block color="falcon-primary" name="" isLoading={isLoading} onClick={handleSaveAndAdd}>
                Save and add another
              </LoaderButton>
            ) : null}
          </FormGroup>
        </Col>
        <Col xs="4">
          <FormGroup>
            <LoaderButton block color="falcon-primary" onClick={handleSave} name="save-and-add" isLoading={isLoading}>
              Save
            </LoaderButton>
          </FormGroup>
        </Col>
      </Row>

      {productId && vendorId === 'hpe' && (
        <Row style={{ textAlign: 'center', marginTop: '20px' }} form>
          <Col lg="12">
            <FormGroup>
              <Button color="falcon-danger" style={{ textAlign: 'center' }} onClick={handleDelete}>
                <FontAwesomeIcon icon={faTrash} /> Delete Managed Product
              </Button>
            </FormGroup>
          </Col>
        </Row>
      )}
    </>
  );
}

ProductActions.propTypes = {
  snProductOptions: PropTypes.array,
  isLoading: PropTypes.bool,
  handleDelete: PropTypes.func,
  handleSave: PropTypes.func,
  handleSaveAndAdd: PropTypes.func,
  vendorId: PropTypes.string,
  productId: PropTypes.string
};
