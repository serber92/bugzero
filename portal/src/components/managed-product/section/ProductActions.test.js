import React from 'react';
import { shallow } from 'enzyme';
import ProductActions from './ProductActions';
test('should test ProductActions component', () => {
  const wrapper = shallow(
    <ProductActions
      vendorId="cisco"
      snProductOptions={[]}
      isLoading={false}
      handleDelete={() => {}}
      handleSave={() => {}}
      handleSaveAndAdd={() => {}}
      productId={'1'}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
