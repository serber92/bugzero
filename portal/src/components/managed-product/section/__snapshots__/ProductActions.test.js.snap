// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test ProductActions component 1`] = `
<Fragment>
  <Row
    className="flex-between-center"
    style={
      Object {
        "marginTop": "40px",
      }
    }
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="4"
    >
      <FormGroup
        tag="div"
      >
        <Button
          block={true}
          color="falcon-danger"
          onClick={[Function]}
          tag="button"
        >
          Cancel
        </Button>
      </FormGroup>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="4"
    >
      <FormGroup
        tag="div"
      />
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="4"
    >
      <FormGroup
        tag="div"
      >
        <LoaderButton
          block={true}
          color="falcon-primary"
          isLoading={false}
          name="save-and-add"
          onClick={[Function]}
        >
          Save
        </LoaderButton>
      </FormGroup>
    </Col>
  </Row>
</Fragment>
`;
